File: client/android/app/src/main/java/app/mobiling/client/catalog/CatalogFeatureBridge.kt
package app.mobiling.client.catalog

import app.mobiling.client.contract.catalog.browse.CatalogNodeSummary
import app.mobiling.client.contract.catalog.browse.ListCatalogNodesQuery
import app.mobiling.client.contract.catalog.detail.CatalogNodeDetailPayload
import app.mobiling.client.data.catalog.browse.CatalogBrowseGateway
import app.mobiling.client.data.catalog.detail.CatalogNodeDetailGateway
import app.mobiling.client.usecase.catalog.browse.ListCatalogNodesUseCase
import app.mobiling.client.usecase.catalog.detail.LoadCatalogNodeDetailUseCase

/**
 * Marketing America Corp. Oleksandr Tishchenko
 *
 * App-level bridge for catalog-domain controlled rewire.
 * It preserves a stable catalog feature entry point while delegating
 * behavior into canonical Contract/Data/UseCase slices.
 */
class CatalogFeatureBridge(
    private val browseGateway: CatalogBrowseGateway,
    private val detailGateway: CatalogNodeDetailGateway,
) {
    suspend fun list(query: ListCatalogNodesQuery): List<CatalogNodeSummary> =
        ListCatalogNodesUseCase(browseGateway).invoke(query)

    suspend fun detail(nodeId: String): CatalogNodeDetailPayload? =
        LoadCatalogNodeDetailUseCase(detailGateway).invoke(nodeId)
}

File: client/ios/App/Catalog/CatalogFeatureBridge.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
// App-level bridge for catalog-domain controlled rewire.
public struct CatalogFeatureBridge: Sendable {
    private let browseGateway: any CatalogBrowseGateway
    private let detailGateway: any CatalogNodeDetailGateway

    public init(
        browseGateway: any CatalogBrowseGateway,
        detailGateway: any CatalogNodeDetailGateway
    ) {
        self.browseGateway = browseGateway
        self.detailGateway = detailGateway
    }

    public func list(query: ListCatalogNodesQuery) async throws -> [CatalogNodeSummary] {
        try await ListCatalogNodesUseCase(gateway: browseGateway)(query: query)
    }

    public func detail(nodeId: String) async throws -> CatalogNodeDetailPayload? {
        try await LoadCatalogNodeDetailUseCase(gateway: detailGateway)(nodeId: nodeId)
    }
}
